src/engine/paris.md

```diff
@@ -111,9 +111,11 @@ Payload validation process consists of validating a payload with respect to the
       payload satisfying the above conditions.
   * Client software **MUST NOT** surface an `INVALID` payload over any API endpoint and p2p interface.
 
-4. Client software **MAY** provide additional details on the validation error if a payload is deemed `INVALID` by assigning the corresponding message to the `validationError` field.
+4. Client software **MUST** not classify the same payload as either both `INVALID` and `VALID` or `INVALID_BLOCK_HASH` and `VALID`.
```

Comment:
What if we put it this way:

4. Validation process **MUST** be idempotent with respect to the payload status. That is, if `engine_newPayload` or `engine_forkchoiceUpdated` responded with either `VALID` or `INVALID | INVALID_BLOCK_HASH` payload status, a future call to these methods **MUST** keep returning that status given the same payload is passed onto the call.

Reply:
This is probably clear enough about handling `SYNCING | ACCEPTED`. As far as I can tell both per spec and practically, it's okay for an EL to vacillate between either `INVALID | INVALID_BLOCK_HASH` and `SYNCING | ACCEPTED` (e.g., because it doesn't have a reference state accessible to check the block against anymore), and similarly between `SYNCING | ACCEPTED` and `VALID`.

So this create a larger change, perhaps a worthwhile one (it's a little weird to see a CL return `VALID` on a block then `SYNCING` on the same block 10 seconds later, and maybe the spec shouldn't allow that), but it is a change, moreso than the wording as written so far.

I'm not opposed to either though, and in many ways, your wording makes more for a more useful spec.

Reply:
I think changing from `VALID` to `NOT_VALIDATED` may only happen is some edge cases and I guess ELs if store a block into their DB can conclude it is valid even though validation process can’t be run due to lack of e.g. a pre-state. We can also append the following to clarify this:

Additionally, `VALID`, `INVALID`, `INVALID_BLOCK_HASH` are considered as the end statuses that **MUST** never change once concluded.

(i am not sure that *concluded* term would be the right one to use in this context, happy to replace it with something more suitable)

Reply:
From the [ACDE#175](https://github.com/ethereum/pm/issues/901) call: `INVALID -> SYNCING` status transition should be allowed because different sync strategies employed by EL clients may not support `INVALID` to be an end state.

Reply:
And there was some preference to make this explicit, as I understand it.